peekingduck/pipeline/nodes/dabble/fps.py

```diff
@@ -72,20 +72,22 @@ def run(self, inputs: Dict[str, Any]) -> Dict[str, Any]:
         frame_fps = 1./(curr_frame_timestamp - self.prev_frame_timestamp)
         self.prev_frame_timestamp = curr_frame_timestamp
 
-        # Calculate global cumulative moving average
-        self.global_avg_fps = self._global_average(frame_fps)
-
         # Calculate moving average FPS (dampen_fps)
         average_fps = self._moving_average(frame_fps)
 
-        if self.fps_log_display:
-            if self.count%self.fps_log_freq == 0:
-                self.logger.info('Avg FPS over last 10 frames: %.2f', average_fps)
+        # Ignore FPS of final frame to avoid skewing final average
+        if not inputs["pipeline_end"]:
+            if self.fps_log_display:
+                if self.count % self.fps_log_freq == 0:
```

Comment:
the current change does not take into consideration the initial count at 0. this is because 0 % self.fps_log_freq is == 0. It passes the test and thus returns a super low ave fps as there is actually only one frame:
<img width="837" alt="Screenshot 2021-08-13 at 9 41 37 AM" src="https://user-images.githubusercontent.com/47622049/129293320-0407e081-f870-4007-9f76-a6c848322e4c.png">

By adding this, the 1st frame ave fps issue is removed:
<img width="837" alt="Screenshot 2021-08-13 at 9 44 03 AM" src="https://user-images.githubusercontent.com/47622049/129293387-0d7c72c0-dd0a-40b9-bcf7-fc11033dd396.png">


```suggestion
                if self.count % self.fps_log_freq == 0 and self.count != 0:
```

Reply:
This is the output if you show the log for every frame. The first 10 frames will be a bit lesser 
```
2021-08-13T09:55:40 peekingduck.declarative_loader INFO:Config for node dabble.fps is updated to: 'fps_log_freq': 1
2021-08-13T09:55:40 peekingduck.pipeline.nodes.dabble.fps INFO:FPS will be logged every: 1 frames
2021-08-13T09:55:43 peekingduck.pipeline.nodes.dabble.fps INFO:Avg FPS over last 10 frames: 0.12
2021-08-13T09:55:43 peekingduck.pipeline.nodes.dabble.fps INFO:Avg FPS over last 10 frames: 7.20
2021-08-13T09:55:43 peekingduck.pipeline.nodes.dabble.fps INFO:Avg FPS over last 10 frames: 9.54
2021-08-13T09:55:43 peekingduck.pipeline.nodes.dabble.fps INFO:Avg FPS over last 10 frames: 10.69
2021-08-13T09:55:43 peekingduck.pipeline.nodes.dabble.fps INFO:Avg FPS over last 10 frames: 11.38
2021-08-13T09:55:43 peekingduck.pipeline.nodes.dabble.fps INFO:Avg FPS over last 10 frames: 11.70
2021-08-13T09:55:43 peekingduck.pipeline.nodes.dabble.fps INFO:Avg FPS over last 10 frames: 11.87
2021-08-13T09:55:43 peekingduck.pipeline.nodes.dabble.fps INFO:Avg FPS over last 10 frames: 12.09
2021-08-13T09:55:43 peekingduck.pipeline.nodes.dabble.fps INFO:Avg FPS over last 10 frames: 12.19
2021-08-13T09:55:43 peekingduck.pipeline.nodes.dabble.fps INFO:Avg FPS over last 10 frames: 12.34
2021-08-13T09:55:43 peekingduck.pipeline.nodes.dabble.fps INFO:Avg FPS over last 10 frames: 13.64
```

Reply:
Good observation and I have made the change